pgxn/neon/pagestore_client.h

```diff
@@ -298,4 +298,46 @@ extern void set_cached_relsize(NRelFileInfo rinfo, ForkNumber forknum, BlockNumb
 extern void update_cached_relsize(NRelFileInfo rinfo, ForkNumber forknum, BlockNumber size);
 extern void forget_cached_relsize(NRelFileInfo rinfo, ForkNumber forknum);
 
+/*
+ * Relation kind enum.
+ */
+typedef enum
+{
+    /* The persistence is not known */
+	RELKIND_UNKNOWN,
+
+    /* The relation is a permanent relation that is WAL-logged normally */
+	RELKIND_PERMANENT,
+
+    /* The relation is an unlogged table/index, stored only on local disk */
+	RELKIND_UNLOGGED,
+
+    /*
+     * The relation is a permanent (index) relation, but it is being built by an in-progress
+     * transaction. It currently only lives on local disk and hasn't been WAL-logged yet.
+     * It will turn into a permanent relation later when the index build completes.
+     * This is currently used for GiST, SP-GiST and GIN indexes, as well as the pgvector
+     * extension.
+     */
+	RELKIND_UNLOGGED_BUILD
+} RelKind;
```

Comment:
"relkind" is a bad term because it already means something else (`pg_class.relkind`)

Reply:
Search & replace will do the rename it in 5 minutes :-)

I'd suggest `RelPersistence`. I don't think we need to shorten it further. 

This is called 'relpersistence' throughout Postgres code. The states we have here are a little different though. Postgres defines:

```
#define           RELPERSISTENCE_PERMANENT      'p' /* regular table */
#define           RELPERSISTENCE_UNLOGGED       'u' /* unlogged permanent table */
#define           RELPERSISTENCE_TEMP           't' /* temporary table */
```

We don't distinguish TEMP and UNLOGGED here, and we have the extra UNLOGGED_BUILD state. It would actually be good to refer to those upstream states in a comment here, explaining the differences.

Maybe `NeonRelPersistence` to distinguish from those upstream states?